homeassistant/components/shelly/__init__.py

```diff
@@ -39,16 +47,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         entry.data.get(CONF_PASSWORD),
         temperature_unit,
     )
-
-    if COAP_CONTEXT not in hass.data[DOMAIN]:
-        coap_context = hass.data[DOMAIN][
-            COAP_CONTEXT
-        ] = await aiocoap.Context.create_client_context()
-        hass.data[DOMAIN][UNDO_SHUTDOWN_LISTENER] = hass.bus.async_listen(
-            EVENT_HOMEASSISTANT_STOP, shutdown_listener(hass)
-        )
-    else:
-        coap_context = hass.data[DOMAIN][COAP_CONTEXT]
+    coap_context = hass.data[DOMAIN][GENERAL][COAP_CONTEXT]
```

Comment:
You shut down the context if the last config entry is unloaded. So that means that here you could have no longer a coap context.

Reply:
I'm ashamed to admit, but I thought `async_setup` is called again when the user configures a device, after removing all previous ones. I haven't checked it before 🤦